src/lib/encoder/Encoder.h

```diff
@@ -55,12 +56,27 @@ inline unsigned Encoder::encode_fountain(const std::string& filename, const std:
 		requiredFrames = 1;
 
 	unsigned i = 0;
+	unsigned consecutiveScansFailed = 0;
 	while (i < requiredFrames)
 	{
 		auto frame = encode_next(*fes, canvas_size);
 		if (!frame)
 			break;
 
+		// some % of generated frames (for the current 8x8 impl)
+		// will produce random patterns that falsely match as
+		// corner "anchors" and fail to extract. So:
+		// if frame fails the scan, skip it.
+		if (!Scanner::will_it_scan(*frame))
+		{
+			if (++consecutiveScansFailed < 5)
+				continue;
+
+			// else, we gotta make forward progress. And it's probably a bug?
+			std::cerr << fmt::format("generated {} bad frames in a row. This really shouldn't happen, maybe report a bug. :(", consecutiveScansFailed) << std::endl;
```

Comment:
I expect that the likely way we hit this message is that the Scanner code gets messed up and everything (or many things) fail to scan. We could risk the infinite loop since that kind of failure should get caught by tests/etc, but... this seems smarter to me.